### DISCUSSION TITLE
again about vscode auto import extension.

### DISCUSSION THREAD

**Discussion by magic-thomas** [adonisjs/core#3885] [Help]
I made User User Model / User Factory 

```sh
    node ace make:model User 
    node ace make:factory User 
```

for example in routes.ts

```js
import User from 'App/Models/User';

Route.get('/', async ({ view }) => {
    const users = await User.all() ; // no problem of auto import.

    UserFactory.create();  // can not load import namespace automatically.
})

```

I can automatically insert  'import User from 'App/Models/User' with cursor on 'User' with 'typescript importer' extension.

But I can not import UserFactory with namespace  automatically. 

It says 

```sh
    no importable symbols found!
```

auto importing function is very important for fast typing. 

Can somebody help me ?

**Comment by RomainLanz** [comment]
Hey @magic-thomas! 👋🏻 

Ensure your editor is correctly installed, and TypeScript Language Server is activated.
This is built-in. There is no need for any third-party extension.

**Reply by magic-thomas** [reply]
I don't get it. 

I installed 'TypeScript Language Server'  by 'npm install -g typescript-language-server typescript'  and reloaded vscode.

I wrote that code again. but no any magic happened. 

How can I check if typescript-language-server  is activate ?

How do you insert 'import' codes automatically on top of ts file ?

I am from Laravel, not familiar with modern node js tools.

**Reply by RomainLanz** [reply]
I am talking about your IDE, not your application.

`TypeScript Language Server` is something built-in [your IDE](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide).

**Reply by magic-thomas** [reply]
Could you answer this plz, if you know how ?

`
How can I check if typescript-language-server is activated ?

How do you insert 'import' codes automatically on top of ts file ?
`

My IDE is working ok, so could be installed correctly. 

I understand <code>TypeScript Language Server</code> is built-in in VSCode. that's fine. 

But I don't develop IDE. 

I just want to know how to insert import namespace and Class with hot key.

**Reply by thetutlage** [reply]
None of this is related to AdonisJS. I think asking your questions in VSCode discord or forum will have better chance of getting a solution
